[GTK][WPE][WebXR] Add the XR session permission request

fujii opened the pull request:
#### fcc54d1d1bb4aa3942544f4fbeb6f6d51f00a660
<pre>
[GTK][WPE][WebXR] Add the XR session permission request
<a href="https://bugs.webkit.org/show_bug.cgi?id=297450">https://bugs.webkit.org/show_bug.cgi?id=297450</a>

Reviewed by Carlos Garcia Campos.

Added a new API for XR session permission request.

Co-authored-by: Carlos Garcia Campos &lt;cgarcia@igalia.com&gt;

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/WebKitUIClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitXRPermissionRequest.cpp: Added.
(webkitXRPermissionRequestAllow):
(webkitXRPermissionRequestDeny):
(webkit_permission_request_interface_init):
(webkitXRPermissionRequestDispose):
(webkit_xr_permission_request_class_init):
(webkit_xr_permission_request_get_security_origin):
(webkit_xr_permission_request_get_session_mode):
(toWebKitXRSessionMode):
(webkitXRPermissionRequestCreate):
* Source/WebKit/UIProcess/API/glib/WebKitXRPermissionRequest.h.in: Added.
* Source/WebKit/UIProcess/API/glib/WebKitXRPermissionRequestPrivate.h: Added.
* Source/WebKit/UIProcess/API/glib/webkit.h.in:
* Tools/MiniBrowser/gtk/BrowserTab.c:
(webKitXRSessionModeToString):
(decidePermissionRequest):

Canonical link: <a href="https://commits.webkit.org/298944@main">https://commits.webkit.org/298944@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d33e1103ade41fdf5e2e334f9a845c179c78c66

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/117273 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36943 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/27556 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/123372 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/69258 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119148 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37639 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45530 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/88993 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/43670 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120206 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/29955 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/105146 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/69500 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-18-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/23260 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67045 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/99354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/23442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/126493 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44170 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33179 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97664 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44526 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/101382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/97459 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/42812 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/20738 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/40520 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18712 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44043 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/49702 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43499 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46844 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45195 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Tools/MiniBrowser/gtk/BrowserTab.c:
```diff
@@ -321,6 +321,21 @@ static void permissionRequestDialogResponse(GtkWidget *dialog, gint response, Pe
     g_clear_pointer(&requestData, permissionRequestDataFree);
 }
 
+static const gchar *WebKitXRSessionModeToString(WebKitXRSessionMode mode)
```
```suggestion
static const gchar *webKitXRSessionModeToString(WebKitXRSessionMode mode)
```

fujii committed:
[GTK][WPE][WebXR] Add the XR session permission request
https://bugs.webkit.org/show_bug.cgi?id=297450

Reviewed by Carlos Garcia Campos.

Added a new API for XR session permission request.

Co-authored-by: Carlos Garcia Campos <cgarcia@igalia.com>

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/API/glib/WebKitUIClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitXRPermissionRequest.cpp: Added.
(webkitXRPermissionRequestAllow):
(webkitXRPermissionRequestDeny):
(webkit_permission_request_interface_init):
(webkitXRPermissionRequestDispose):
(webkit_xr_permission_request_class_init):
(webkit_xr_permission_request_get_security_origin):
(webkit_xr_permission_request_get_session_mode):
(toWebKitXRSessionMode):
(webkitXRPermissionRequestCreate):
* Source/WebKit/UIProcess/API/glib/WebKitXRPermissionRequest.h.in: Added.
* Source/WebKit/UIProcess/API/glib/WebKitXRPermissionRequestPrivate.h: Added.
* Source/WebKit/UIProcess/API/glib/webkit.h.in:
* Tools/MiniBrowser/gtk/BrowserTab.c:
(webKitXRSessionModeToString):
(decidePermissionRequest):

Canonical link: https://commits.webkit.org/298944@main
